### PR TITLE
Qute - add TemplateExtension#matchNames()

### LINKED ISSUE #25994: Qute - support comma-separated list of values in TemplateExtension#matchName()
### Description

Currently, you can use `TemplateExtension#matchRegex()` but a list of values is a very common use case and the regex used at runtime is much less performant (and unnecessary complex).

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #25994

### CODE REVIEW

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 1]
```diff
@@ -1591,6 +1588,61 @@ This template extension method makes it possible to render the following templat
...
 <1> `item` is resolved to an instance of `org.acme.Item`.
 
+However, it is possible to specify the matching name with `matchName()`.
+
+.`TemplateExtension#matchName()` Example
+[source,java]
+----
+@TemplateExtension(matchName = "discounted")
+static BigDecimal discountedPrice(Item item) {
+   // this method matches {item.discounted} iff "item" resolves to an object assignable to "Item"
```
Comment (gastaldi) [inline docs/src/main/asciidoc/qute-reference.adoc:1598]:
```suggestion
   // this method matches {item.discounted} if "item" resolves to an object assignable to "Item"
```

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 2]
```diff
@@ -1591,6 +1588,61 @@ This template extension method makes it possible to render the following templat
...
+----
+
+A special constant - `TemplateExtension#ANY`/`*` - can be used to specify that the extension method matches any name.
+
+.`TemplateExtension#ANY` Example
+[source,java]
+----
+@TemplateExtension(matchName = "*")
+static String itemProperty(Item item, String name) { <1>
+   // this method matches {item.foo} iff "item" resolves to an object assignable to "Item"
```
Comment (gastaldi) [inline docs/src/main/asciidoc/qute-reference.adoc:1610]:
```suggestion
   // this method matches {item.foo} if "item" resolves to an object assignable to "Item"
```

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 3]
```diff
@@ -1591,6 +1588,61 @@ This template extension method makes it possible to render the following templat
...
+<1> A additional string method parameter is used to pass the actual property name.
+
+It's also possible to match the name against a regular expression specified in `matchRegex()`.
+
+.`TemplateExtension#matchRegex()` Example
+[source,java]
+----
+@TemplateExtension(matchRegex = "foo|bar")
+static String itemProperty(Item item, String name) { <1>
+   // this method matches {item.foo} and {item.bar} iff "item" resolves to an object assignable to "Item"
```
Comment (gastaldi) [inline docs/src/main/asciidoc/qute-reference.adoc:1623]:
```suggestion
   // this method matches {item.foo} and {item.bar} if "item" resolves to an object assignable to "Item"
```

File: docs/src/main/asciidoc/qute-reference.adoc [hunk 4]
```diff
@@ -1591,6 +1588,61 @@ This template extension method makes it possible to render the following templat
...
+Finally, `matchNames()` can be used to specify a collection of matching names. 
+An additional string method parameter is mandatory as well.
+
+
+.`TemplateExtension#matchNames()` Example
+[source,java]
+----
+@TemplateExtension(matchNames = {"foo", "bar"})
+static String itemProperty(Item item, String name) {
+   // this method matches {item.foo} and {item.bar} iff "item" resolves to an object assignable to "Item"
```
Comment (gastaldi) [inline docs/src/main/asciidoc/qute-reference.adoc:1638]:
```suggestion
   // this method matches {item.foo} and {item.bar} if "item" resolves to an object assignable to "Item"
```

File: docs/src/main/asciidoc/qute-reference.adoc (hunk 1 above)
Comment (mkouba) [reply docs/src/main/asciidoc/qute-reference.adoc:1598]: Well, I meant https://en.wikipedia.org/wiki/IFF...

Comment (gastaldi) [reply docs/src/main/asciidoc/qute-reference.adoc:1598]: That looked like a typo to me 😀 

I'd suggest using the longer `if (and only if)` for clarity

**Review by gastaldi — APPROVED** [review]

**Pull request merged**